mysql-test/suite/tianmu/t/create_table.test

```diff
@@ -0,0 +1,336 @@
+--error 1146
+create table test.t1 like x;
+drop table if exists test.t1;
+#
+#Bogus error message on attempt to CREATE TABLE t LIKE view
+#
+create database mysqltest;
+use mysqltest;
+create view v1 as select 'foo' from dual;
+--error 1347
+create table t1 like v1;
+drop view v1;
+drop database mysqltest;
+#
+#SET value having comma not correctly handled
+#
+use test;
+--error 1367
+create table t1(a set("a,b","c,d") not null);
+#
+#mysqld segfault when trying to create table with same name as existing view
+#
+CREATE TABLE t1 (a INT);
+CREATE TABLE t2 (a INT);
+INSERT INTO t1 VALUES (1),(2),(3);
+INSERT INTO t2 VALUES (1),(2),(3);
+CREATE VIEW v1 AS SELECT t1.a FROM t1, t2;
+--error 1050
+CREATE TABLE v1 AS SELECT * FROM t1;
+DROP VIEW v1;
+DROP TABLE t1,t2;
+SET sql_mode = 'NO_ZERO_DATE';
+CREATE TABLE t2(c1 TIMESTAMP, c2 TIMESTAMP DEFAULT 0);
+DROP TABLE t2;
+CREATE TABLE t2(c1 TIMESTAMP NULL);
+SET sql_mode = '';
+DROP TABLE t2;
+#
+CREATE TABLE t1(c1 YEAR DEFAULT 2008, c2 YEAR DEFAULT 0);
+SHOW CREATE TABLE t1;
+INSERT INTO t1 VALUES();
+SELECT * FROM t1;
+drop table t1;
+#
+# Attempt to create table with invalid default should fail in normal mode
+#
+--error 1067
+create table t1 (dt datetime default '2008-02-31 00:00:00');
+set @old_mode= @@sql_mode;
+set @@sql_mode='ALLOW_INVALID_DATES';
+--echo # The same should be possible in relaxed mode
+create table t1 (dt datetime default '2008-02-31 00:00:00');
+set @@sql_mode= @old_mode;
+--error 1067
+create table t2 like t1;
+set @@sql_mode='ALLOW_INVALID_DATES';
+create table t2 like t1;
+show create table t1;
+show create table t2;
+set @@sql_mode= @old_mode;
+drop tables t1,t2;
+#
+# Creating a base table in presence of an updatable view.
+# 
+create table t2 (a int unique);
+create view t1 as select a from t2;
+insert into t1 (a) values (1);
+--error 1050
+create table t1 (a int);
+create table if not exists t1 (a int);
+--error 1050
+create table t1 (a int) select 2 as a;
+select * from t1;
+create table if not exists t1 (a int) select 2 as a;
+select * from t1;
+select * from t2;
+create temporary table if not exists t1 (a int) select 3 as a;
+select * from t1;
+select * from t2;
+drop table t1,t2;
+drop view t1;
+#
+# Repeating the test with a view select a constant number
+#
+create view t1 as select 1 as a;
+--error 1471
+insert into t1 (a) values (1);
+--error 1288
+update t1 set a=3 where a=2;
+--error 1050
+create table t1 (a int);
+create table if not exists t1 (a int);
+--error 1050
+create table t1 (a int) select 2 as a;
+select * from t1;
+# Note: an essential warning.
+create table if not exists t1 (a int) select 2 as a;
+select * from t1;
+create temporary table if not exists t1 (a int) select 3 as a;
+select * from t1;
+# Note: an essential warning.
+create temporary table if not exists t1 (a int) select 4 as a;
+select * from t1;
+drop temporary table t1;
+drop view t1;
+# Test cases added for coverage.
+CREATE TABLE t1(fld1 DATETIME NOT NULL DEFAULT '1111:11:11');
```

Comment:
SQL statements use the same style, including uppercase keywords, lowercase other variables, table names, column names, etc